BoxContentSDK/BoxContentSDK/Categories/UIApplication+ExtensionSafeAdditions.m

```diff
@@ -8,37 +8,65 @@
 
 @implementation UIApplication (ExtensionSafeAdditions)
 
-+ (UIApplication *)box_sharedApplication
++ (BOOL)box_isRunningExtension
 {
-#ifndef BOX_TARGET_IS_EXTENSION
-    // If we are compiling from a non-extension target, use the regular sharedApplication.
-    return [UIApplication sharedApplication];
-#else
-    // If we are compiling from an extension, return nil here since UIApplication isn't extension-safe.
-    return nil;
-#endif
+    return [[[NSBundle mainBundle] executablePath] containsString:@".appex/"];
```

Comment:
we wil go this direction as it seems to be the more reliable until we can syncup with apple engineers at WWDC about a possible better approach.